### PR TITLE
chore: prepare release v.4.0.0-rc.0

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,55 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+
+- feat(logs): add `sumologic.logs.additionalFields` property [#3286]
+- feat(metrics): add additionalServiceMonitors setting [#3292]
+
+### Changed
+
+- feat(prometheus): Removing prometheus recording rules [#3211]
+- feat(metrics): move app metrics filtering to metadata layer [#3232]
+- feat!: use OTLP sources by default [#3297]
+- feat!(metrics): move extra processors after sumologic_schema [#3306]
```
Comment (swiatekm-sumo) [inline CHANGELOG.md:34]: These should be under breaking changes. I think we mislabelled the changelog files.

Comment (aboguszewski-sumo) [reply CHANGELOG.md]: There were also some breaking changes under `Fixed` section, I moved them as well.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,55 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+
+- feat!: remove support for fluent-bit and fluentd [#3244]
+- feat!: truncate fullname after 22 characters [#3248]
+- feat(metrics)!: use otel by default [#3284]
+- feat!: use OTLP sources by default [#3297]
+- feat!(metrics): move extra processors after sumologic_schema [#3306]
+- fix(metrics)!: drop k8s.node.name attribute [#3295]
+- fix(logs)!: move JSON parsing after user-defined processors [#3281] The log body will now always be a string if accessed in extra
+  processors. Users who want to access specific fields in their parsed JSON log should explicitly call ParseJSON in their processor
+  definition.
```
Comment (swiatekm-sumo) [inline CHANGELOG.md:28]:
```suggestion
- fix(logs)!: move JSON parsing after user-defined processors [#3281] 

  The log body will now always be a string if accessed in extra
  processors. Users who want to access specific fields in their parsed JSON log should explicitly call ParseJSON in their processor
  definition.
```

**Review by swiatekm-sumo — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (aboguszewski-sumo) [reply CHANGELOG.md]: @swiatekm-sumo Prettier doesn't like your suggestion. The best we can do is:

```
- fix(logs)!: move JSON parsing after user-defined processors [#3281]

  The log body will now always be a string if accessed in extra processors. Users who want to access specific fields in their parsed JSON
  log should explicitly call ParseJSON in their processor definition.
```

**Pull request merged**